### DISCUSSION TITLE
api call to fimwaremanagement always fails

### DISCUSSION THREAD

**Discussion by hamza-outa** [AxisCommunications/discussions#172] [VAPIX]
I am trying to change the firmware by using the API but it always returns a 400.
I came across this discussion: https://github.com/orgs/AxisCommunications/discussions/123
when I try this curl it fails with this error:
```
{
  "apiVersion": "1.4",
  "error": {
    "code": 400,
    "message": "Invalid JSON format."
  }
}
```

this is the curl command I use:
`curl --anyauth -u "root:pswrd" -F 'body={"apiVersion":"1.0","context":"my-context","method":"upgrade"}' -F packfil=@P3265-LVE_10_12_130.bin "192.168.0.3/axis-cgi/firmwaremanagement.cgi"`

I also tried changing the apiVersion to 1.4, but to no avail

**Comment by mattias-kindborg-at-work** [comment]
Hi @hamza-outa. I don't see any issues with your curl command. Which firmware version is currently installed on the Axis device?

**Reply by hamza-outa** [reply]
firmware on the camera: 10.12.130
I also tried upgrading the firmware to 11.2.28 but gives the same error

**Comment by vivekatoffice** [comment]
Related link:
[123](https://github.com/orgs/AxisCommunications/discussions/123)
